contrib/mesos/pkg/executor/service/service.go

```diff
@@ -173,6 +173,19 @@ func (s *KubeletExecutorServer) runKubelet(
 		kcfg.DockerDaemonContainer = "" // don't move the docker daemon into a cgroup
 		kcfg.Hostname = kcfg.HostnameOverride
 		kcfg.KubeClient = apiclient
+
+		// taken from KubeletServer#Run(*KubeletConfig)
+		clientConfig, err := s.CreateAPIServerClientConfig()
+		if err == nil {
+			kcfg.KubeClient, err = client.New(clientConfig)
```

Comment:
i think you can just reuse `apiclient` here instead of creating a new client.Client. the bits you really need are below, creating the `eventClientConfig` and an `EventClient`.